src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramDescriptionFormField/DeclaredProgramDescriptionFormField.test.ts

```diff
@@ -0,0 +1,183 @@
+import { DECLARED_PROGRAM_DESCRIPTION_MAX_LENGTH } from '@/features/student/declaredPrograms/config'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const DeclaredProgramDescriptionTextareaStub = defineComponent({
+  name: 'DeclaredProgramDescriptionTextarea',
+  props: {
+    modelValue: String,
+    errorMessage: String
+  },
+  emits: ['update:modelValue', 'blur'],
+  template: `
+    <div data-testid="declared-program-description-textarea-stub">
+      <textarea
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+        @blur="$emit('blur')"
+      />
+      <slot name="customCaptions" />
+    </div>
+  `
+})
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramDescriptionFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        description: ''
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit ({ value }) {
+          if (!value.description || value.description.trim() === '') {
+            return {
+              fields: {
+                description: 'La description est requise'
+              }
+            }
+          }
+          return { fields: { description: undefined } }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramDescriptionFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program description form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    DeclaredProgramDescriptionTextarea: DeclaredProgramDescriptionTextareaStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the description textarea component', () => {
+      const textarea = wrapper.findComponent({ name: 'DeclaredProgramDescriptionTextarea' })
+      expect(textarea.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have empty initial value', () => {
+      const textarea = wrapper.findComponent({ name: 'DeclaredProgramDescriptionTextarea' })
+      expect(textarea.props('modelValue')).toBe('')
+    })
+  })
+
+  BddTest().and('the user enters a description', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+      const textarea = wrapper.findComponent({ name: 'DeclaredProgramDescriptionTextarea' })
+      const textareaElement = textarea.find('textarea')
+      await textareaElement.setValue('This is my program description')
+      await wrapper.vm.$nextTick()
+    })
```

Comment:
```suggestion
  BddTest().and('the user enters a description', () => {
      const textarea = wrapper.findComponent({ name: 'DeclaredProgramDescriptionTextarea' })
      const textareaElement = textarea.find('textarea')
      await textareaElement.setValue('This is my program description')
      await wrapper.vm.$nextTick()
    })
```

Il faut mettre ce and dans le when et dans ce cas-là t'as plus besoin de tes clearAllMocks et mount. Ce commentaire est valable pour tous les and je pense